### DISCUSSION TITLE
Default root path with $locale

### DISCUSSION THREAD

**Discussion by bitdom8** [ivanhofer/typesafe-i18n#326] [Q&A] [answered]
Instead of coding each time 

`<a href="/{$locale}/about">{$t('menu.about')}</a>`

Can we introduce our API one time our root URL and pages can have this $locale value by default

`<a href="/about">{$t('menu.about')}</a>`

Will still direct to `<a href="/{$locale}/about">{$t('Menu')}</a>`

So our hrefs takes the root as http://localhost:3000/de or http://localhost:3000/it , just given an instance

**Comment by ivanhofer** [comment, marked as answer]
This is something that is not related to `typesafe-i18n` itself.

In your example you would need to tell SvelteKit how to handle those links.
1. You could create your own `<A>` component that accepts an "href" and prepends it with the correct locale.
2. Store the locale information in a session cookie and read the information from there instead reading it from the url.
3. You could write a global script that replaces all "hrefs" in the current view and adds the locale (could be a bit hacky).
4. Create a global redirect in your `hooks` file and redirect to the language specific url if no locale information is present (not recommended).
5. Or some other option.

**Reply by bitdom8** [reply]
Thanks, I think going with $locale each time is more clear:)
